Cleanup Travis and tox configs

moggers87 opened the pull request:
Cleaned up the Travis and tox configs, including some bits missed in my PR to django-sudo:

* Moved the environment list completely to tox, which should be easier to read given how we need to exclude quite a few Django/Python combinations
* Added linting and doc building to tox and fixed a few issues
* Linting and doc building are done as separate jobs on Travis
* Kept all this compatible with the already defined `invoke` tasks
* Removed dependencies from `dev-requirements.txt` that are no longer needed

moggers87 committed:
Keep tox and travis in sync

…by keeping the envlist in tox and using tox-travis to run tests on the
CI. A lot more concise than that long list of excludes we had before!

moggers87 committed:
Fix warning from pytest

moggers87 committed:
Add sphinx to travis

moggers87 committed:
Fix sphinx warnings

moggers87 committed:
Flake8 should be run by travis too

moggers87 committed:
Fixup minor flake8 error

moggers87 committed:
Removed dependencies that are no longer required

flake8 and sphinx will be installed by tox when required

justinmayer commented on dev-requirements.txt:
```diff
@@ -1,9 +1,6 @@
 invoke>=0.11.1,<0.12
-pytest
-pytest-cov
-pytest-django
```
Do you think we need to remove the `pytest` packages here? Sometimes it's useful to do a quick test run on the current Python interpreter. If you still think we should remove these, then the reference to `py.test` should probably be removed from the [Contributing > Running Tests](https://django-elevate.readthedocs.io/en/latest/contributing/index.html#running-tests) docs.

moggers87 replied:
I think I'll add it back in. I've since tried to do things like tox -e py27 and found that it does not work as I had assumed.

justinmayer replied:
Okay, cool. Might consider doing the same with `flake8`, since we presumably want to encourage folks to use it. 😉

justinmayer reviewed (commented):
Nice work, Matt. The switch to `tox-travis` is a big improvement. Other than my question about `py.test`, this all looks good. Looking forward to merging it.

moggers87 committed:
Add back flake8 and pytest to dev-requirements.txt

Running tox can be cumbersome, so allow developers to use flake8 and
pytest outside of tox.

Added django to dev-requirements.txt too otherwise pytest will error.

moggers87 commented:
I've added flake8 and pytest back into dev-requirements.txt. I've also added django to that file because pytest is pretty useless without it.

justinmayer commented:
Looks good to me. Do you think any of these commits should be squashed? I'm happy to combine them into a single commit upon merge, of course... Just wasn't sure if you wanted to organize them in any particular semantic manner. ☺️

moggers87 commented:
I have no personal preference, so feel free to do a squash-merge ☺

justinmayer commented:
Many thanks for enhancements, Matt. 👍
